### PR TITLE
move progress bar examples

### LINKED ISSUE #825: Example: progress
Split appropriate existing examples into separate files using functional approach and icache and move into the correct `/src/examples/widgets` subdirectory.

### PR DESCRIPTION
**Pull request by vansimke**
**Type:** bug / feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit tests are included in the PR
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

**Description:** Moved progress bar examples to new format and structure

Resolves #825

### CODE REVIEW

File: src/examples/src/widgets/progress/ProgressWithChangingValues.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { icache } from '@dojo/framework/core/middleware/icache';
+import Progress from '@dojo/widgets/progress';
+import Button from '@dojo/widgets/Button';
```
Comment (JamesLMilner) [inline src/examples/src/widgets/progress/ProgressWithChangingValues.tsx]: I think this is failing because this should be ` '@dojo/widgets/button';`

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-e7pzeo6qe.now.sh
* Tests Deployment: https://dojowidgets-qubvteomk.now.sh

File: src/examples/src/widgets/progress/ProgressWithChangingValues.tsx (hunk 1 above)
Comment (vansimke) [reply src/examples/src/widgets/progress/ProgressWithChangingValues.tsx]: Thanks for catching that, @JamesLMilner . I've fixed the import and rebased onto latest master. Care to take another look?

File: src/examples/src/config.tsx [hunk 2]
```diff
@@ -103,6 +103,10 @@ import CloseableDialog from './widgets/dialog/CloseableDialog';
 
 `!has('docs')`;
 import testsContext from './tests';
+import ProgressWithChangingValues from './widgets/progress/ProgressWithChangingValues';
```
Comment (tomdye) [inline src/examples/src/config.tsx]: mike can you move these up above the `!has('docs')` line please and alphabetise them like the others? This helps with conflicts later

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-435ojiq3q.now.sh
* Tests Deployment: https://dojowidgets-2hi74veez.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-8gu5zjq40.now.sh
* Tests Deployment: https://dojowidgets-g4gom875v.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-8gu5zjq40.now.sh
* Tests Deployment: https://dojowidgets-g4gom875v.now.sh

File: src/examples/src/widgets/progress/ProgressWithChangingValues.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export default factory(function ProgressWithChangingvalues({ middleware: { icache } }) {
+	const max = 100;
+	const step = 10;
+	const value = icache.getOrSet<number>('value', 0);
+	return (
+		<div>
+			<Progress value={value} max={max} />
+			<div>
+				<Button onClick={() => icache.set('value', value - step < 0 ? 0 : value - step)}>
```
Comment (matt-gadd) [inline src/examples/src/widgets/progress/ProgressWithChangingValues.tsx]:
```suggestion
				<Button onClick={() => icache.set('value', Math.max(0, value - step))}>
```

File: src/examples/src/widgets/progress/ProgressWithChangingValues.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+					Decrease
+				</Button>
+				<Button
+					onClick={() => icache.set('value', value + step > max ? max : value + step)}
```
Comment (matt-gadd) [inline src/examples/src/widgets/progress/ProgressWithChangingValues.tsx]:
```suggestion
					onClick={() => icache.set('value', Math.min(value - step, max))}
```

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-sb6o6gcfd.now.sh
* Tests Deployment: https://dojowidgets-ixunxjzgv.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-5pm5mmvg4.now.sh
* Tests Deployment: https://dojowidgets-2y7il2a8g.now.sh

**Pull request merged**